pkg/backup/compaction_dist_test.go

```diff
@@ -0,0 +1,209 @@
+			},
+			numNodes: 1,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("b")}},
+			},
+		},
+		{
+			name: "single node, multiple span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+				{Span: roachpb.Span{Key: Key("c"), EndKey: Key("d")}},
+				{Span: roachpb.Span{Key: Key("e"), EndKey: Key("f")}},
+			},
+			numNodes: 1,
+			expectedDistribution: [][]roachpb.Span{
+				{
+					{Key: Key("a"), EndKey: Key("b")},
+					{Key: Key("c"), EndKey: Key("d")},
+					{Key: Key("e"), EndKey: Key("f")},
+				},
+			},
+		},
+		{
+			name: "single node, multiple overlapping span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+				{Span: roachpb.Span{Key: Key("b"), EndKey: Key("d")}},
+				{Span: roachpb.Span{Key: Key("e"), EndKey: Key("f")}},
+			},
+			numNodes: 1,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("d")}, {Key: Key("e"), EndKey: Key("f")}},
+			},
+		},
+		{
+			name: "multiple nodes, single span entry",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+			},
+			numNodes: 3,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("b")}},
+				nil,
+				nil,
+			},
+		},
+		{
+			name: "multiple nodes, exact multiple of span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+				{Span: roachpb.Span{Key: Key("c"), EndKey: Key("d")}},
+				{Span: roachpb.Span{Key: Key("e"), EndKey: Key("f")}},
+				{Span: roachpb.Span{Key: Key("g"), EndKey: Key("h")}},
+				{Span: roachpb.Span{Key: Key("i"), EndKey: Key("j")}},
+				{Span: roachpb.Span{Key: Key("k"), EndKey: Key("l")}},
+			},
+			numNodes: 3,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("b")}, {Key: Key("c"), EndKey: Key("d")}},
+				{{Key: Key("e"), EndKey: Key("f")}, {Key: Key("g"), EndKey: Key("h")}},
+				{{Key: Key("i"), EndKey: Key("j")}, {Key: Key("k"), EndKey: Key("l")}},
+			},
+		},
+		{
+			name: "multiple nodes, exact multiple of overlapping span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+				{Span: roachpb.Span{Key: Key("b"), EndKey: Key("d")}},
+				{Span: roachpb.Span{Key: Key("e"), EndKey: Key("f")}},
+				{Span: roachpb.Span{Key: Key("f"), EndKey: Key("h")}},
+				{Span: roachpb.Span{Key: Key("i"), EndKey: Key("j")}},
+				{Span: roachpb.Span{Key: Key("j"), EndKey: Key("l")}},
+			},
+			numNodes: 3,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("d")}},
+				{{Key: Key("e"), EndKey: Key("h")}},
+				{{Key: Key("i"), EndKey: Key("l")}},
+			},
+		},
+		{
+			name: "multiple nodes, not exact multiple of span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
```

Comment:
could you use 2 2 1 instead of 3 3 2? just trying to reduce the lines of code a reader needs to read.